src/threads/thread.cpp

```diff
@@ -825,7 +832,11 @@ int hpxc_setspecific(hpxc_key_t key, const void* value)
 
 void* hpxc_getspecific(hpxc_key_t key)
 {
-    thread_handle* self = ::get_thread_data(hpx::threads::get_self_id());
+    hpx::threads::thread_id_type id = hpx::threads::get_self_id();
+    // Check if this is a valid hpx thread
+    if (id == hpx::threads::invalid_thread_id)
+        return nullptr;
```

Comment:
Also I don't really like this being here, but I had a kmp call to `hpxc_getspecific` from outside the HPX runtime, so this was the only way to solve it. 
@light2802 Do you think this `__kmp_internal_end_dtor` could be made to run from within an HPX thread?